### PR TITLE
Support building RHEL-8 Azure VHD and SIG Images

### PR DESCRIPTION
**Pull request by kkeshavamurthy**
What this PR does / why we need it:
Add support for building RHEL-8 Azure VHD & SIG Images. 

Which issue(s) this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged): Fixes #

**Additional context**
Add any other context for the reviewers

### CODE REVIEW

**Comment by kkeshavamurthy** [comment]
@jsturtevant @mboersma This should be good to go. PTAL.

File: images/capi/packer/azure/centos-7-gen2.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "build_name": "centos-7-gen2",
   "distribution": "centos",
   "distribution_release": "centos-7",
-  "distribution_version": "7.7",
+  "distribution_version": "7",
```
Comment (mboersma) [inline images/capi/packer/azure/centos-7-gen2.json:5]: This seems unrelated to the RHEL 8 changes. Is this so the build will use the latest 7.x version (currently CentOS 7.9)?

Comment (kkeshavamurthy) [reply images/capi/packer/azure/centos-7-gen2.json:5]: Right. `distribution_version` is used in image definition names, goss, ansible conditions etc. So I standardized it to `7` instead of `7.7` since it makes more sense. It'll also help in running the right tests in goss. `image_sku` is used to pick the right base images from the azure marketplace and it's left at `7.7`.

File: images/capi/packer/azure/centos-7.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "build_name": "centos-7",
   "distribution": "centos",
   "distribution_release": "centos-7",
-  "distribution_version": "7.7",
+  "distribution_version": "7",
   "image_offer": "CentOS",
   "image_publisher": "OpenLogic",
   "image_sku": "7.7"
```
[the remark this answers is not in the document]
Comment (kkeshavamurthy) [reply images/capi/packer/azure/centos-7.json:8]: `image_sku` is not linked to `distribution_version` AFAIK.

**Review by mboersma — COMMENTED** [review]
/lgtm
